compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/IsolationAnalyzer.java

```diff
@@ -419,12 +425,16 @@ public void visit(BLangFunction funcNode) {
 
         if (this.inferredIsolated &&
                 !isIsolated(symbol.flags) &&
-                !Symbols.isFlagOn(symbol.flags, Flags.WORKER) &&
                 functionIsolationInferenceInfo != null &&
                 functionIsolationInferenceInfo.dependsOnlyOnInferableConstructs &&
                 // Init method isolation depends on the object field-initializers too, so we defer inference.
                 !funcNode.objInitFunction) {
             functionIsolationInferenceInfo.inferredIsolated = true;
+            if (Symbols.isFlagOn(symbol.flags, Flags.WORKER)) {
+                symbol.flags |= Flags.ISOLATED;
+                funcNode.getBType().flags |= Flags.ISOLATED;
+                funcNode.getBType().tsymbol.flags |= Flags.ISOLATED;
```

Comment:
Why do we need to do this specifically for workers here? Don't we set them at the end anyway?